src/otx/core/types/device.py

```diff
@@ -0,0 +1,21 @@
+"""OTX Device type definition."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from enum import Enum
+
+
+class DeviceType(str, Enum):
+    """OTX Device type definition."""
+    # ("cpu", "gpu", "tpu", "ipu", "hpu", "mps", "auto")
+
+    auto = "auto"
+    gpu = "gpu"
+    cpu = "cpu"
+    tpu = "tpu"
+    ipu = "ipu"
+    hpu = "hpu"
+    mps = "mps"
```

Comment:
How that list was generated, and while Intel devices (xpu) are missing?

Reply:
This is a list of accelerators currently supported by lightning.Trainer.